language/diem-framework/modules/0L/Subsidy.move

```diff
@@ -49,26 +50,73 @@ address 0x1 {
       if (subsidy_units > len && subsidy_units > 0 ) { // arithmetic safety check
         subsidy_granted = subsidy_units/len;
       } else { return };
-
+      
       let i = 0;
       while (i < len) {
-        let node_address = *(Vector::borrow<address>(outgoing_set, i));
+        let node_address = Vector::borrow<address>(outgoing_set, i);
         // Transfer gas from vm address to validator
-        let minted_coins = Diem::mint<GAS>(vm, subsidy_granted);
-        DiemAccount::vm_deposit_with_metadata<GAS>(
-          vm,
-          node_address,
-          minted_coins,
-          b"validator subsidy",
-          b""
-        );
-
+        check_team_and_pay(vm, node_address, subsidy_granted);
         // refund operator tx fees for mining
-        refund_operator_tx_fees(vm, node_address);
+        refund_operator_tx_fees(vm, *node_address);
         i = i + 1;
       };
     }
 
+    fun check_team_and_pay(vm: &signer, captain_address: &address, subsidy_granted: u64) {
+      // this is a solo validator. Exists during transition to delegation mode. This is a fallback condition to keep the node from halting
+      let captain_value = subsidy_granted;
+      if (Teams::team_is_init(*captain_address)) {
+        // split captain reward and send to captain.
+        let captain_pct = Teams::get_operator_reward(*captain_address);
+        // split off the captain value
+        captain_value = FixedPoint32::multiply_u64(
+          subsidy_granted,
+          FixedPoint32::create_from_rational(captain_pct, 100) 
+        );
+        let value_to_members = subsidy_granted - captain_value;
+        // get team members
+        let members = Teams::get_team_members(*captain_address);
+        // split the team subsidy
+        split_subsidy_to_team(vm, &members, value_to_members);
+      };
+      let captain_coins = Diem::mint<GAS>(vm, captain_value);
+            
+      // payment to captain
+      DiemAccount::vm_deposit_with_metadata<GAS>(
+        vm,
+        *captain_address,
+        captain_coins,
+        b"validator subsidy",
+        b""
+      );
+    }
+
+    public fun split_subsidy_to_team(vm: &signer, members: &vector<address>, value_to_members: u64) {
```

Comment:
this is a public function, maybe we need some checks

Reply:
Thanks good catch. Let me take a look.

Reply:
fixed with c60c0c213606987d3e537b533af6bd62c1c850d4